corehq/apps/locations/models.py

```diff
@@ -664,6 +673,7 @@ def path(self, value):
         self._path = value
 
     @classmethod
+    @quickcache(['location_id'], timeout=30 * 60)
```

Comment:
I also don't think this is a good cache target without really robust invalidation.  With this cached, any calls to `SQLLocation.by_location_id` will return data stale by up to 30 minutes if the location object is changed.